hitbox-tarantool/src/backend.rs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        self.client()?
+            .eval(
+                "
+                    local space_name = ...
+                    box.schema.space.create(space_name, { if_not_exists = true })
+                    box.space[space_name]:create_index('primary', {
+                        parts = { { 1, 'string' } },
+                        if_not_exists = true,
+                    })
+
+                    if not _G.__hitbox_cache_fiber then
+                        _G.__hitbox_cache_fiber = require('fiber').create(function()
+                            local fiber = require('fiber')
+                            fiber.name('hitbox_cache_fiber')
+                            while true do
+                                local ok, res = pcall(function()
+                                    for _, t in box.space[space_name]:pairs() do
+                                        if t[2] <= fiber.time() then
+                                            box.space[space_name]:delete(t[1])
+                                        end
+                                    end
+                                end)
+
+                                if not ok then
+                                    require('log').error(err)
+                                end
+
+                                fiber.testcancel()
+                                fiber.sleep(1)
+                            end
+                        end)
+                    end
+                ",
+                &("hitbox_cache".to_string(),),
+            )
+            .await
+            .map_err(|err| BackendError::InternalError(Box::new(err)))?;
+
+        Ok(())
+    }
+
+    fn map_err(err: Error) -> BackendError {
+        BackendError::InternalError(Box::new(err))
+    }
+
+    async fn call<T>(&self, cmd: &str, params: &T) -> BackendResult<Vec<CacheEntry>>
+    where
+        T: Serialize,
+    {
+        let result: Vec<CacheEntry> = self
+            .client()?
+            .prepare_fn_call(format!("box.space.hitbox_cache:{}", cmd))
+            .bind_ref(params)
+            .map_err(TarantoolBackend::map_err)?
+            .execute()
+            .await
+            .map_err(TarantoolBackend::map_err)?
+            .decode_result_set()
+            .map_err(TarantoolBackend::map_err)?;
+
+        Ok(result)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct CacheEntry {
+    pub key: String,
+    pub ttl: Option<u32>,
+    pub value: String,
+}
+
+#[async_trait]
+impl CacheBackend for TarantoolBackend {
+    async fn get<T>(&self, key: String) -> BackendResult<Option<CachedValue<T::Cached>>>
+    where
+        T: CacheableResponse,
+        <T as CacheableResponse>::Cached: serde::de::DeserializeOwned,
+    {
+        let entries = self.call("get", &(key)).await?;
+        let entry = match entries.first() {
+            Some(v) => Some(
+                JsonSerializer::<String>::deserialize(v.value.clone())
```

Comment:
I would suggest using BinSerializer instead of JsonSerializer. BinSerializer works more efficiently with bytes.
https://github.com/hit-box/hitbox/blob/tower/hitbox-backend/src/serializer.rs#L112

Reply:
OK, but we'll do it later :)

Reply:
It should be fixed in new Backend trait implementation.
Serialization logic will be split into two methods `get` and `get_raw` and moved into default implementation of  `get`.